test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,640 @@
+	return wait.PollUntil(5*time.Second, func() (bool, error) {
+		var (
+			currentOp *compute.Operation
+			err       error
+		)
+
+		if op.Region != "" {
+			region := getResourceNameFromSelfLink(op.Region)
+			currentOp, err = computeService.RegionOperations.Get(project, region, op.Name).Context(ctx).Do()
+		} else {
+			currentOp, err = computeService.GlobalOperations.Get(project, op.Name).Context(ctx).Do()
+		}
+
+		if err != nil {
+			return false, err
+		}
+		return currentOp.Status == "DONE", nil
+	}, ctx.Done())
+}
+
+func getResourceNameFromSelfLink(link string) string {
+	parts := strings.Split(link, "/")
+	return parts[len(parts)-1]
+}
+
+func createBastion(cluster *controller.Cluster, name string, project string) (*extensionsv1alpha1.Bastion, *bastionctrl.Options) {
+	bastion := &extensionsv1alpha1.Bastion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name + "-bastion",
+			Namespace: name,
+		},
+		Spec: extensionsv1alpha1.BastionSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type: "gcp",
+			},
+			UserData: []byte(userDataConst),
+			Ingress: []extensionsv1alpha1.BastionIngressPolicy{
+				{IPBlock: networkingv1.IPBlock{
+					CIDR: myPublicIP,
+				}},
+			},
+		},
+	}
+
+	options, err := bastionctrl.DetermineOptions(bastion, cluster, project)
+	Expect(err).NotTo(HaveOccurred())
+
+	return bastion, options
+}
+
+func createInfrastructureConfig() *gcpv1alpha1.InfrastructureConfig {
+	return &gcpv1alpha1.InfrastructureConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: gcpv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "InfrastructureConfig",
+		},
+		Networks: gcpv1alpha1.NetworkConfig{
+			Workers: workersSubnetCIDR,
+		},
+	}
+}
+
+func createShootStruct(infrastructureConfig []byte) *gardencorev1beta1.Shoot {
+
+	shoot := &gardencorev1beta1.Shoot{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "core.gardener.cloud/v1beta1",
+			Kind:       "Shoot",
+		},
+		Spec: gardencorev1beta1.ShootSpec{
+			Region: *region,
+			Provider: gardencorev1beta1.Provider{
+				InfrastructureConfig: &runtime.RawExtension{
+					Raw: []byte(infrastructureConfig),
+				}},
+		},
+	}
+	return shoot
+}
+
+func createCloudProfile() *gardencorev1beta1.CloudProfile {
+	cloudProfile := &gardencorev1beta1.CloudProfile{
+		Spec: gardencorev1beta1.CloudProfileSpec{
+			Regions: []gardencorev1beta1.Region{
+				{Name: *region},
+				{Name: *region, Zones: []gardencorev1beta1.AvailabilityZone{
+					{Name: *region + "-b"},
+					{Name: *region + "-c"},
+				}},
+			},
+		},
+	}
+	return cloudProfile
+}
+
+func createClusters(name string) (*extensionsv1alpha1.Cluster, *controller.Cluster) {
+	infrastructureConfig := createInfrastructureConfig()
+	infrastructureConfigJSON, _ := json.Marshal(&infrastructureConfig)
+
+	shoot := createShootStruct(infrastructureConfigJSON)
+
+	shootJSON, _ := json.Marshal(shoot)
+
+	CloudProfile := createCloudProfile()
+
+	CloudProfileJSON, _ := json.Marshal(CloudProfile)
```

Comment:
```suggestion
	cloudProfile := createCloudProfile()
	cloudProfileJSON, _ := json.Marshal(cloudProfile)
```

Reply:
fixed 783eb84

Reply:
I don't think this comment is fixed. It basically suggest to make the vars camel case.